pkg/runtime/codec.go

```diff
@@ -16,11 +16,47 @@ limitations under the License.
 
 package runtime
 
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/yaml"
+)
+
 // CodecFor returns a Codec that invokes Encode with the provided version.
 func CodecFor(scheme *Scheme, version string) Codec {
 	return &codecWrapper{scheme, version}
 }
 
+// yamlCodec converts YAML passed to the Decoder methods to JSON.
+type yamlCodec struct {
+	// a Codec for JSON
+	Codec
+}
+
+// yamlCodec implements Codec
+var _ Codec = yamlCodec{}
```

Comment:
To prove that yamlCodec implements Codec.  We started doing that in other places where there's a type that *should* implement a named interface.

----- Original Message -----
> >  // CodecFor returns a Codec that invokes Encode with the provided version.
> >  func CodecFor(scheme *Scheme, version string) Codec {
> >  	return &codecWrapper{scheme, version}
> >  }
> >  
> > +// yamlCodec converts YAML passed to the Decoder methods to JSON.
> > +type yamlCodec struct {
> > +	// a Codec for JSON
> > +	Codec
> > +}
> > +
> > +// yamlCodec implements Codec
> > +var _ Codec = yamlCodec{}
> 
> What's the point of this line?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/5533/files#r26594752